modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/servlet/taglib/BaseDisplayTag.java

```diff
@@ -129,6 +154,8 @@ protected Map<String, Object> prepareProps(Map<String, Object> props) {
 			props.put("additionalProps", _additionalProps);
 		}
 
+		props.put("namespace", getNamespace());
```

Comment:
Hey @markocikos, I don't know what namespace is being used for on FDS but If this should be unique for every FDS displayed on the page. I think we should append a string like UNNAMED_FDS or UNREGISTRED_FDS.

Reply:
A random string was used in the previous version, not sure if it's the way...

Do we need the `_randomNamespace` variable and related methods now?

Reply:
`namespace` does not need to be unique for every FDS. `id` needs to be unique, nothing else. If there is something inside FDS instance that needs to generate a unique value, it should probably be a derivative of `id`.

`namespace` refers to **portlet** namespace. This is consistent over basically all tags in DXP. We can easily have multiple FDS instances in a portlet, as portlets tend to have multiple pages.

Within a DXP tag, namespace is typically required to
1. Build parameters for DXP requests
2. Access DOM elements in a portlets that have been set with `<portlet:namespace />`

Looking at master, `randomNamespace` is not used anywhere :thinking: I think we can safely deprecate it.

Reply:
On a closer inspection, we are using `randomNamespace` [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/servlet/taglib/BaseDisplayTag.java#L128). Admittedly, I don't see a purpose in setting `id` of this `div`, but let's leave this for a separate task.